docs/siem/timeline/timeline-templates.asciidoc

```diff
@@ -0,0 +1,107 @@
+<<create-rule-ui>>.
+
+When you load {es-sec} prebuilt rules, some prebuilt Timeline templates are
+also loaded. You can attach these templates to detection rules and use them as
+the basis of your own custom templates. The following templates are loaded:
+
+* `Generic Endpoint Timeline`: Useful for investigating Elastic Endpoint
+security alerts.
+* `Generic Process Timeline`: Useful for investigating process-related alerts.
+* `Generic Network Timeline`: Useful for investigating network-related alerts. 
+
+[discrete]
+=== Create a Timeline template
+
+. Go to *Security* -> *Timelines*.
+. Click the *Templates* tab, and then *Create new timeline template*.
++
+[role="screenshot"]
+image::images/create-template-ui.png[]
+
+. Give the template a title.
+. Optionally, add a description and notes.
+. To add template fields and queries, click *Add field*, and then select the
+required option:
+
+* _Add field_: Add a query (field and field value).
+* _Add template field_: Add a field without a predetermined value.
+
+TIP: You can also drag and send items to the template from the *Overview*,
+*Hosts*, *Network*, and *Detections* pages.
+
+*Example*
+
+To create a template for process-related alerts on a specific host:
+
+* Add an 'ordinary' query for the host name:
+`host.name: "Linux_staforshire-061"`
+* Add template field for process names: `process.name: "{process.name}"`
+
+[role="screenshot"]
+image::images/template-query-example.png[]
+
+When alerts generated by rules associated with this template are investigated
+in Timeline, the host name is `Linux_staforshire-061`, whereas the process name
+value is retrieved from the alert's `process.name` field.
+
+[discrete]
+=== Manage existing Timeline templates
+
+You can view, duplicate, delete, and create templates from existing Timelines:
+
+. Go to *Security* -> *Timelines*.
+. Click the *Templates* tab.
+. Click the More actions icon in the relevant row, and then select the action:
+
+* _Create timeline from template_
+* _Duplicate template_
+* _Delete template_
+
+TIP: To perform the same action on multiple templates, select templates and
+then the required action from the _Bulk actions_ menu.
+
+NOTE: You cannot delete prebuilt templates.
+
+[discrete]
+=== Export and import Timeline templates
+
+You can import and export Timeline templates, which enables importing templates
+from one {kib} space or instance to another. Exported templates are saved in an
+http://ndjson.org[`ndjson`] file.
+
+. Go to *Security* -> *Timelines*.
+. Click the *Templates* tab.
+. To export templates, do one of the following:
+
+* To export one template, click the More actions icon in the relevant row and
+then select _Export selected_.
+* To export multiple templates, select all the required templates and then click
+*Bulk actions* -> _Export selected_.
+
+. To import templates, click *Import Timeline* and then select or drap-and-drop
+the template `ndjson` file.
```

Comment:
could we put a note here to let them know the content of the ndjson has to be in minimised format to import it properly.

e.g. 

This is ok
```
{"savedObjectId":"67664480-d191-11ea-ae67-4f4be8c1847b","version":"WzU1NSwxXQ==","columns":[{"indexes":null,"name":null,"columnHeaderType":"not-filtered","id":"@timestamp","searchable":null},{"indexes":null,"name":null,"columnHeaderType":"not-filtered","id":"message","searchable":null},{"indexes":null,"name":null,"columnHeaderType":"not-filtered","id":"event.category","searchable":null},{"indexes":null,"name":null,"columnHeaderType":"not-filtered","id":"event.action","searchable":null},{"indexes":null,"name":null,"columnHeaderType":"not-filtered","id":"host.name","searchable":null},{"indexes":null,"name":null,"columnHeaderType":"not-filtered","id":"source.ip","searchable":null},{"indexes":null,"name":null,"columnHeaderType":"not-filtered","id":"destination.ip","searchable":null},{"indexes":null,"name":null,"columnHeaderType":"not-filtered","id":"user.name","searchable":null}],"dataProviders":[],"description":"","eventType":"all","filters":[],"kqlMode":"filter","timelineType":"default","kqlQuery":{"filterQuery":{"serializedQuery":"{\"bool\":{\"should\":[{\"exists\":{\"field\":\"@timestamp\"}}],\"minimum_should_match\":1}}","kuery":{"expression":"@timestamp : * ","kind":"kuery"}}},"title":"x2","sort":{"columnId":"@timestamp","sortDirection":"desc"},"created":1596036895488,"createdBy":"angela","updated":1596491470411,"updatedBy":"angelachuang","templateTimelineId":null,"templateTimelineVersion":null,"dateRange":{"start":"2020-04-10T14:10:58.373Z","end":"2020-05-30T14:16:58.373Z"},"savedQueryId":null,"eventNotes":[{"noteId":"7d875ba0-d5d3-11ea-9899-ebec3d084fe0","version":"WzU1NiwxXQ==","eventId":"8KtMKnIBOS_moQ_K9fAe","note":"hi Xavier","timelineId":"67664480-d191-11ea-ae67-4f4be8c1847b","created":1596491490806,"createdBy":"angelachuang","updated":1596491490806,"updatedBy":"angelachuang"}],"globalNotes":[],"pinnedEventIds":["K99zy3EBDTDlbwBfpf6x","GKpFKnIBOS_moQ_Ke5AO","8KtMKnIBOS_moQ_K9fAe"]}
```

This will fail
```
{
  "savedObjectId": "67664480-d191-11ea-ae67-4f4be8c1847b",
  "version": "WzU1NSwxXQ==",
  "columns": [
    {
      "indexes": null,
      "name": null,
      "columnHeaderType": "not-filtered",
      "id": "@timestamp",
      "searchable": null
    },
    {
      "indexes": null,
      "name": null,
      "columnHeaderType": "not-filtered",
      "id": "message",
      "searchable": null
    },
    {
      "indexes": null,
      "name": null,
      "columnHeaderType": "not-filtered",
      "id": "event.category",
      "searchable": null
    },
    {
      "indexes": null,
      "name": null,
      "columnHeaderType": "not-filtered",
      "id": "event.action",
      "searchable": null
    },
    {
      "indexes": null,
      "name": null,
      "columnHeaderType": "not-filtered",
      "id": "host.name",
      "searchable": null
    },
    {
      "indexes": null,
      "name": null,
      "columnHeaderType": "not-filtered",
      "id": "source.ip",
      "searchable": null
    },
    {
      "indexes": null,
      "name": null,
      "columnHeaderType": "not-filtered",
      "id": "destination.ip",
      "searchable": null
    },
    {
      "indexes": null,
      "name": null,
      "columnHeaderType": "not-filtered",
      "id": "user.name",
      "searchable": null
    }
  ],
  "dataProviders": [],
  "description": "",
  "eventType": "all",
  "filters": [],
  "kqlMode": "filter",
  "timelineType": "default",
  "kqlQuery": {
    "filterQuery": {
      "serializedQuery": "{\"bool\":{\"should\":[{\"exists\":{\"field\":\"@timestamp\"}}],\"minimum_should_match\":1}}",
      "kuery": {
        "expression": "@timestamp : * ",
        "kind": "kuery"
      }
    }
  },
  "title": "x2",
  "sort": {
    "columnId": "@timestamp",
    "sortDirection": "desc"
  },
  "created": 1596036895488,
  "createdBy": "angela",
  "updated": 1596491470411,
  "updatedBy": "angelachuang",
  "templateTimelineId": null,
  "templateTimelineVersion": null,
  "dateRange": {
    "start": "2020-04-10T14:10:58.373Z",
    "end": "2020-05-30T14:16:58.373Z"
  },
  "savedQueryId": null,
  "eventNotes": [
    {
      "noteId": "7d875ba0-d5d3-11ea-9899-ebec3d084fe0",
      "version": "WzU1NiwxXQ==",
      "eventId": "8KtMKnIBOS_moQ_K9fAe",
      "note": "hi Xavier",
      "timelineId": "67664480-d191-11ea-ae67-4f4be8c1847b",
      "created": 1596491490806,
      "createdBy": "angelachuang",
      "updated": 1596491490806,
      "updatedBy": "angelachuang"
    }
  ],
  "globalNotes": [],
  "pinnedEventIds": [
    "K99zy3EBDTDlbwBfpf6x",
    "GKpFKnIBOS_moQ_Ke5AO",
    "8KtMKnIBOS_moQ_K9fAe"
  ]
}
```